app/file-detail/controller.js

```diff
@@ -69,10 +83,60 @@ export default Controller.extend({
             window.location = url;
         },
 
-        changeView() {
-            $('#mfrIframeParent').toggle();
-            $('#revisionsPanel').toggle();
-            $('.view-button').toggleClass('btn-default btn-primary');
+        delete() {
+            this.get('model.file').destroyRecord()
+                .then(() => this._deleteFile('success'))
+                .catch(() => this._deleteFile('error'));
+        },
+
+        showModal() {
+            this.set('openModal', true);
+        },
+
+        closeModal() {
+            this.set('openModal', false);
+        },
+
+        changeViewPanel(panel, button) {
+            if (!this.get('isEditableFile')) {
+                $('#mainViewBtn, #revisionBtn').toggleClass('btn-primary btn-default');
+                $('#mfrIframeParent, #revisionsPanel').toggle();
+                return;
+            }
+
+            if (button === 'revisionBtn' && $(`#${panel}`).css('display') === 'none') {
+                $('.panel-view').hide().removeClass('col-sm-6');
+                $('.view-button').removeClass('btn-primary').addClass('btn-default');
+                $(`#${panel}`).toggle();
+                $(`#${button}`).toggleClass('btn-default btn-primary');
+                return;
+            } else if (button === 'revisionBtn') {
+                $('#mfrIframeParent').toggle();
+                $('#mainViewBtn').toggleClass('btn-default btn-primary');
+                $(`#${panel}`).toggle();
+                $(`#${button}`).toggleClass('btn-default btn-primary');
+                return;
+            }
```

Comment:
this would be clearer as:
```
if (button === 'revisionBtn') {
    if ($(`#${panel}`).css('display') === 'none') {
        ...
    } else {
        ...
    }
}
```
Also
```
$(`#${panel}`).toggle();
$(`#${button}`).toggleClass('btn-default btn-primary');
```
appears in both cases so it could go in the outer if block